Connect to mysql databases

wardsimon opened the pull request:
Simple edit to allow for connection to remote and local mysql databases

wardsimon committed:
Updated to connect to mysql databases

wardsimon committed:
Port needs to be numeric...

eugenio412 commented:
Could you do it again but this time with a working setting for SQlite? Because I had to revert it because it wasn't working with the ordinary database

wardsimon commented:
Hey,

Yes, I realised that there would be problems as soon as I pressed the pull request. I'll explain what I did, so you can have a check.
- Added PyMySQL==0.7.9 to requirements.txt as it does not ship normally 
- Imported PyMySQL as re (to parse the mysql string)
- Added a line to the config-bot.json which is 'false' for a local db and a string in the format mysql://user:pass@host:port/database
- Changed the connect to database section to read the preference.
- If mysql, parse the config file and then connect using pymysql. This has the advantage that you can use the cur = con.parser(); and not have to change code.

I have tested this with my local db and remote db and it has worked, but please check as I might have been silly. 

I hope this helps. :-)
